test-harness/src/test/java/com/netflix/conductor/tests/integration/AbstractWorkflowServiceTest.java

```diff
@@ -348,6 +348,64 @@ public void testTaskDefTemplate() throws Exception {
+    public void testKafkaTaskDefTemplate() throws Exception {
+        System.setProperty("STACK_KAFKA", "test_kafka_topic");
+        TaskDef templatedTask = new TaskDef();
+        templatedTask.setName("templated_kafka_task");
+
+        Map<String, Object> kafkaRequest = new HashMap<>();
+        kafkaRequest.put("topic", "${STACK_KAFKA}");
+        kafkaRequest.put("bootStrapServers", "localhost:9092");
+
+        Map<String, Object> value = new HashMap<>();
+        value.put("inputPaths", Arrays.asList("${workflow.input.path1}", "${workflow.input.path2}"));
+        value.put("requestDetails", "${workflow.input.requestDetails}");
+        value.put("outputPath", "${workflow.input.outputPath}");
+        kafkaRequest.put("value", value);
+        templatedTask.getInputTemplate().put("kafka_request", kafkaRequest);
+        metadataService.registerTaskDef(Arrays.asList(templatedTask));
+
+        WorkflowDef templateWf = new WorkflowDef();
+        templateWf.setName("template_workflow_kafka");
+        WorkflowTask wft = new WorkflowTask();
+        wft.setName(templatedTask.getName());
+        wft.setWorkflowTaskType(TaskType.SIMPLE);
+        wft.setTaskReferenceName("t0");
+        templateWf.getTasks().add(wft);
+        templateWf.setSchemaVersion(2);
+        metadataService.registerWorkflowDef(templateWf);
+
+        Map<String, Object> requestDetails = new HashMap<>();
+        requestDetails.put("key1", "value1");
+        requestDetails.put("key2", 42);
+
+        Map<String, Object> input = new HashMap<>();
+        input.put("path1", "file://path1");
+        input.put("path2", "file://path2");
+        input.put("outputPath", "s3://bucket/outputPath");
+        input.put("requestDetails", requestDetails);
+
+        String id = startOrLoadWorkflowExecution(templateWf.getName(), 1, "testTaskDefTemplate", input, null, null);
+        assertNotNull(id);
+        Workflow workflow = workflowExecutionService.getExecutionStatus(id, true);
+        assertNotNull(workflow);
+        assertTrue(workflow.getReasonForIncompletion(), !workflow.getStatus().isTerminal());
+        assertEquals(1, workflow.getTasks().size());
+        Task task = workflow.getTasks().get(0);
+        Map<String, Object> taskInput = task.getInputData();
+        assertNotNull(taskInput);
+        assertTrue(taskInput.containsKey("kafka_request"));
+        assertTrue(taskInput.get("kafka_request") instanceof Map);
+
+        ObjectMapper om = new ObjectMapper();
+
+        //Use the commented sysout to get the string value
+        //System.out.println(om.writeValueAsString(om.writeValueAsString(taskInput)));
+        String expected = "{\"kafka_request\":{\"topic\":\"test_kafka_topic\",\"bootStrapServers\":\"localhost:9092\",\"value\":{\"requestDetails\":{\"key1\":\"value1\",\"key2\":42},\"outputPath\":\"s3://bucket/outputPath\",\"inputPaths\":[\"file://path1\",\"file://path2\"]}}}";
+        assertEquals(expected, om.writeValueAsString(taskInput));
```

Comment:
Would it be possible to mock out Kafka publish part and run decide until completion in this test?